src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/vedtaksperiode/Opphørsperiode.kt

```diff
@@ -165,16 +168,49 @@ private fun finnOpphørsperioderMellomUtbetalingsperioder(utbetalingsperioder: L
 
 fun Opphørsperiode.tilVedtaksperiodeMedBegrunnelse(
     vedtak: Vedtak,
-): VedtaksperiodeMedBegrunnelser =
-    VedtaksperiodeMedBegrunnelser(
+): VedtaksperiodeMedBegrunnelser {
+    val behandlingsÅrsak = vedtak.behandling.opprettetÅrsak
+
+    return VedtaksperiodeMedBegrunnelser(
         fom = this.periodeFom,
         tom = this.periodeTom,
         vedtak = vedtak,
         type = this.vedtaksperiodetype,
     ).also { vedtaksperiodeMedBegrunnelser ->
+        if (behandlingsÅrsak == BehandlingÅrsak.OVERGANGSORDNING_2024) {
+            settOvergangsordningOpphørBegrunnelser(vedtaksperiodeMedBegrunnelser)
+        } else {
+            settBegrunnelserPåPeriode(vedtaksperiodeMedBegrunnelser)
+        }
+    }
+}
+
+private fun Opphørsperiode.settOvergangsordningOpphørBegrunnelser(
+    vedtaksperiodeMedBegrunnelser: VedtaksperiodeMedBegrunnelser,
+) {
+    val behandlingKategori = vedtaksperiodeMedBegrunnelser.vedtak.behandling.kategori
+
+    if (behandlingKategori == BehandlingKategori.NASJONAL) {
```

Comment:
Her kunne man kanskje vurdert en `when` for å forhindre at andre kategorier blir lagt inn og automatisk blir behandlet som EØS

Reply:
Godt poeng! ⭐